### PR TITLE
update sphinx version in docs-requirements.txt

### LINKED ISSUE #4180: can't build docs due to multiple targets for cross-reference
I can't successfully build the docs using `python setup.py docs` with Sphinx 1.5.5 due to the following error

```
Warning, treated as error:
/home/thomas/gitrepos/coala/coalib/core/DependencyTracker.py:docstring of coalib.core.DependencyTracker.DependencyTracker.check_circular_dependencies:None: WARNING: more than one target found for cross-reference 'CircularDependencyError': coalib.collecting.Dependencies.CircularDependencyError, coalib.core.CircularDependencyError.CircularDependencyError
```

I had a look at the implementation of the two CircularDependenyError classes and they are quite similar, can't they be merged?

### PR DESCRIPTION
**Pull request by thisch**
Closes #4180 

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

### CODE REVIEW

**Comment by thisch** [comment]
I also tried to merge the CircularDependencyError classes (see https://github.com/thisch/coala/commit/e492cfd7f9e1dce221ce753ac8a919e0e9e2fb39), but according to Lasse the core is currently beeing rewritten. Therefore, I didn't create a new PR for this

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/4183/commits/cfa2708d77f0d409af8dfdd06f8f6f7860840a97

File: coalib/core/DependencyTracker.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def check_circular_dependencies(self):
          ...
         coalib.core.CircularDependencyError.CircularDependencyError: ...
 
-        :raises CircularDependencyError:
+        :raises coalib.core.CircularDependencyError:
```
Comment (userzimmermann) [inline coalib/core/DependencyTracker.py]: Shouldn't this be also `coalib.core.CircularDependencyError.CircularDependencyError` then?

**Comment by userzimmermann** [comment]
unack https://github.com/coala/coala/pull/4183/commits/cfa2708d77f0d409af8dfdd06f8f6f7860840a97

File: coalib/core/DependencyTracker.py (hunk 1 above)
Comment (thisch) [inline coalib/core/DependencyTracker.py]: yes, you are right

Comment (thisch) [inline coalib/core/DependencyTracker.py]: I'm wondering that sphinx does not check that the docstrings contain meaningful settings.

Comment (userzimmermann) [inline coalib/core/DependencyTracker.py]: @thisch I'm wondering too :)

**Comment by userzimmermann** [comment]
@thisch Thanks for quick reaction!

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/4183/commits/162b18237242fdf3e6bbbeeee689f1142c0de5d8

**Comment by userzimmermann** [comment]
@rultor merge

**Pull request merged**